e2e/tests/404.spec.ts

```diff
@@ -35,12 +37,134 @@ test('Bad URL redirection to 404 page', async ({page}) => {
 
       // Assert that the response status code is 404
       expect(response.status()).toBe(404);
+
+      // Check page content
+      const errorMessage = page.locator('#error-detailed-message');
+      await expect(errorMessage).toBeVisible();
+      await expect(errorMessage).toContainText(
+        "We couldn't find the page you're looking for.",
+      );
+
+      // Check buttons
+      await expect(page.locator('#error-action-home-btn')).toBeVisible();
+      await expect(page.locator('#error-action-report')).toBeVisible();
+    });
+  }
+});
+
+test('should fetch similar features from API and show results', async ({
+  page,
+}) => {
+  // Test URLs for different feature IDs
+  const badUrls = [
+    {badUrl: `${BASE_URL}/features/badID_1234`, query: 'badID_1234'},
+    {badUrl: `${BASE_URL}/features/g`, query: 'g'}
+  ];
+  const API_BASE_URL =
+    'http://localhost:8080/v1/features?q={query}&page_size=5';
+
+  for (const {badUrl, query} of badUrls) {
+    await test.step(`Testing API response for: ${badUrl}`, async () => {
+      await page.goto(badUrl);
+      await expect(page).toHaveURL(
+        'http://localhost:5555/errors-404/feature-not-found?q=' + query,
+      );
+
+      const featurePageResponse = await page
+        .context()
+        .request.fetch(page.url());
+
+      // Assert that the response status code is 404
+      expect(featurePageResponse.status()).toBe(404);
+
+      // Mock API response for similar features
+      const apiUrl = API_BASE_URL.replace('{query}', query);
+      const response = await page.context().request.get(apiUrl);
+      expect(response.status()).toBe(200);
+
+      const data = await response.json();
+      const hasResults = Array.isArray(data?.data) && data.data.length > 0;
+
+      if (hasResults) {
+        // Show similar features container
+        const similarContainer = page.locator('.similar-features-container');
+        await expect(similarContainer).toBeVisible();
+        await expect(page.locator('.feature-list li')).toHaveCount(
+          data.data.length,
+        );
+
+        // ✅ Click first similar feature
```

Comment:
```suggestion
        // Click first similar feature
```

Could you remove the emojis from here and other comments?